v2/api/dbforpostgresql/customizations/flexible_server_extensions.go

```diff
@@ -79,3 +85,55 @@ func secretsToWrite(obj *postgresql.FlexibleServer) ([]*v1.Secret, error) {
 
 	return collector.Values()
 }
+
+var _ extensions.PreReconciliationChecker = &FlexibleServerExtension{}
+
+// If a flexible server does not have a provisioningState in this set, it will reject any attempt to PUT a new state
+// out of hand; so there's no point in even trying. This is true even if the PUT we're doing will have no effect on
+// the state of the cluster.
+// These are all listed lowercase, so we can do a case-insensitive match.
+var nonBlockingFlexibleServerStates = set.Make(
```

Comment:
minor: fix varname too? This should be `blockingFlexibleServerStates`, right?

Reply:
No. I changed the name to `nonBlockingFlexibleServerStates` because I changed it to the list of non-blocking states - the states that don't block us from trying to reconcile.

Reply:
Clearly I was tired at the end of the day